src/DatePicker/Utilities.elm

```diff
@@ -314,3 +317,69 @@ toUtcDateTimeString datetime =
         ++ ":"
         ++ addLeadingZero (Time.toSecond Time.utc datetime)
         ++ " (UTC)"
+
+
+setDayNotTime : Maybe Posix -> Posix -> Posix
+setDayNotTime prevPickedDT newPickedDT =
+    let
+        ( prevPickedHour, prevPickedMinute ) =
+            Maybe.map (Time.posixToParts Time.utc) prevPickedDT
+                |> Maybe.map (\parts -> ( parts.hour, parts.minute ))
+                |> Maybe.withDefault ( 0, 0 )
+    in
+    Time.posixToParts Time.utc newPickedDT |> (\parts -> { parts | hour = prevPickedHour, minute = prevPickedMinute }) |> Time.partsToPosix Time.utc
+
+
+type HourOrMinute
+    = IsHour Int
```

Comment:
I fell like a type should not be prefixed with `Is`.

Reply:
I would have liked that as well. However, we end up getting a type collision with the Time module's Hour and Minute constructors without it... perhaps we just remove the need for them by separating the functions out as suggested below.